src/index.js

```diff
@@ -189,7 +189,7 @@ function listen(options, callback) {
     }
 
     if (opts.immediate) callback(mq);
```

Comment:
Here the callback would be called with `mq = undefined` so I guess the check should also prevent this from happening.